File: dart/raindrops.dart
class Raindrops {
  String convert(int n) {
    String s = '';
    if (n % 3 == 0) {
      s += 'Pling';
    }
    if (n % 5 == 0) {
      s += 'Plang';
    }
    if (n % 7 == 0) {
      s += 'Plong';
    }
    return s == '' ? '$n' : s;
  }
}

File: ruby/raindrops.rb
class Raindrops
  def self.convert(n)
    s = ''
    for i, t in [[3, 'Pling'], [5, 'Plang'], [7, 'Plong']] do
      if n % i == 0
        s += t
      end
    end
    s == '' ? "#{n}" : s
  end
end
